compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6291,26 +6291,37 @@ private BType getCompletionType(BType collectionType, BLangQueryExpr queryExpr,
         return completionType;
     }
 
-    private BType getNonContextualQueryType(BType staticType, BType basicType) {
-        BType resultType;
+    private BType getNonContextualQueryType(BType staticType, BType basicType, Location pos) {
         switch (basicType.tag) {
             case TypeTags.TABLE:
-                resultType = symTable.tableType;
+                if (types.isAssignable(staticType, symTable.mapAllType)) {
```

Comment:
Why do we need this check?

Reply:
Table types only allow map types as the static type in query If there is no contextually expected type.
Types other than map will give an error.